projects/gloo/pkg/plugins/istio_automtls/helper.go

```diff
@@ -0,0 +1,35 @@
+package istio_automtls
+
+import (
+	"github.com/solo-io/gloo/projects/gloo/constants"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+)
+
+const EnvoyTransportSocketMatch = "envoy.transport_socket_match"
+
+// AddIstioAutomtlsMetadata adds metadata used by the transport_socket_match
+// to select the mTLS transport socket. The Envoy metadata label is added
+// based on the presence of the Istio workload label "security.istio.io/tlsMode=istio".
+func AddIstioAutomtlsMetadata(
+	metadata *envoy_config_core_v3.Metadata,
+	workloadLabels map[string]string,
+	enableAutoMtls bool,
```

Comment:
nit: I know this is a copy/paste of an existing function, so I'm happy to leave it as is. It feels strange to me that we pass a boolean and only perform an action if that's true. It feels like we could just have the function only be called if the enableAutoMtls value is true

Reply:
Idk, I'd rather have "pass the global setting in, let some shared code interpret it" than have that conditional in multiple places, as easy as it may be.